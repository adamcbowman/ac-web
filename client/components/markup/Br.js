import React from 'react'
import PropTypes from 'prop-types'
import { memo } from 'utils/react'
import styles from './Br.css'

Br.propTypes = {
    ribbon: PropTypes.bool,
}

function Br({ ribbon }) {
    const className = ribbon ? 'Ribbon' : 'Main'

    return <div className={styles[className]} />
}

export default memo.static(Br)
