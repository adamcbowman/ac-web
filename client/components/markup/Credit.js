import React, { memo, createElement } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames/bind'
import Dimensions from 'components/Dimensions'
import { useToggle } from 'utils/react'
import styles from './Credit.css'

Credit.propTypes = {
    children: PropTypes.string.isRequired,
    compact: PropTypes.bool,
    top: PropTypes.bool,
}

function Credit({ top, compact, children }) {
    const [on, toggle] = useToggle(false)
    const className = classNames({
        Credit: true,
        Compact: compact,
        Expanded: on,
        Top: top,
    })

    return (
        <span data-label="Credit" className={className} onClick={toggle}>
            {children}
        </span>
    )
}

const Credit = memo(Credit)

export default Object.assign(Credit, {
    Managed(props) {
        return (
            <Dimensions>
                {({ width }) =>
                    createElement(Credit, {
                        ...props,
                        compact: width < MAGIC_MAX_WIDTH_TO_SHOW_COMPACT_CREDIT,
                    })
                }
            </Dimensions>
        )
    },
})

// Constants
const MAGIC_MAX_WIDTH_TO_SHOW_COMPACT_CREDIT = 250
const classNames = classnames.bind(styles)
