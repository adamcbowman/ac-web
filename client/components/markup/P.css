@import '../../styles/colors.css';

.Capped {
    cursor: pointer;
    position: relative;
    margin-bottom: 1.25em;

    &:before {
        content: 'More';
        position: absolute;
        top: auto;
        right: 0;
        bottom: -1.25em;
        left: 0;
        text-align: right;
        color: var(--primary);
    }

    &:after {
        content: 'â€¦';
    }
}
