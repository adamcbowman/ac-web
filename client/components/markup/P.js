import React from 'react'
import { compose, withHandlers, withState } from 'recompose'
import PropTypes from 'prop-types'
import styles from './P.css'

P.propTypes = {
    children: PropTypes.string.isRequired,
    capAt: PropTypes.number,
    title: PropTypes.string,
}

function P({ children = '', capAt = Infinity, title, ...props }) {
    const capped = children.length > capAt
    const className = capped ? 'Capped' : null

    title = capped ? 'Click to read more' : title
    children = capped ? children.substr(0, capAt) : children

    return (
        <p className={styles[className]} {...props}>
            {children}
        </p>
    )
}

export default compose(
    withState('capAt', 'setCapAt', props => props.capAt),
    withHandlers({
        onClick: props => event => {
            event.preventDefault()

            props.setCapAt(Infinity)
        },
    })
)(P)
