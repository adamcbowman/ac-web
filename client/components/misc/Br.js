import React from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import styles from './Br.css'

Br.propTypes = {
    ribbon: PropTypes.bool,
}

function Br({ribbon = false}) {
    return (
        <div styleName={ribbon ? 'Ribbon' : 'Main'}></div>
    )
}

export default CSSModules(Br, styles)
