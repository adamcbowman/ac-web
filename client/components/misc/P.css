@import "../../styles/colors.css";

.Capped {
    cursor: pointer;
    position: relative;
    margin-bottom: 1.25em;

    &:before {
        content: "More";
        position: absolute auto 0 -1.25em 0;
        text-align: right;
        color: var(--primary);
    }
    &:after {
        content: "â€¦";
    }
}
