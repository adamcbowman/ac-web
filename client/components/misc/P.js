import { DOM } from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import {
    compose,
    withHandlers,
    mapProps,
    withState,
    defaultProps,
    setPropTypes,
} from 'recompose'
import styles from './P.css'

export default compose(
    setPropTypes({
        children: PropTypes.string.isRequired,
        capAt: PropTypes.number,
    }),
    defaultProps({
        children: '',
        capAt: Infinity,
    }),
    withState('capAt', 'setCapAt', props => props.capAt),
    mapProps(({ capAt, children, title, ...props }) => {
        const capped = children.length > capAt

        return {
            ...props,
            children: capped ? children.substr(0, capAt) : children,
            styleName: capped ? 'Capped' : null,
            title: capped ? 'Click to read more' : title,
        }
    }),
    withHandlers({
        onClick: props => event => {
            event.preventDefault()

            props.setCapAt(Infinity)
        },
    }),
    CSSModules(styles)
)(DOM.p)
