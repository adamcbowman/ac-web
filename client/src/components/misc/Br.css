@import "../../styles/colors.css";

.Main {
    composes: import Hash from "../../styles/Textures.css";
    display: block;
    height: 8px;
}

.Ribbon {
    composes: Main;

    &:before {
        content: '';
        display: inline-block;
        float: left;
        background-color: var(--brand-primary);
        width: 50px;
        height: 100%;
    }
}
