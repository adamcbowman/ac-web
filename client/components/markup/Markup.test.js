import React from 'react'
import Renderer from 'react-test-renderer'
import Br from './Br'
import P from './P'
import Credit from './Credit'

test('Br component', () => {
    expect(Renderer.create(<Br />)).toMatchSnapshot()
    expect(Renderer.create(<Br ribbon />)).toMatchSnapshot()
})

test('P component', () => {
    expect(Renderer.create(<P>Some content...</P>)).toMatchSnapshot()
    expect(
        Renderer.create(<P capAt={10}>Some content truncated...</P>)
    ).toMatchSnapshot()
})

test('Credit component', () => {
    expect(Renderer.create(<Credit>Some credit</Credit>)).toMatchSnapshot()
    expect(
        Renderer.create(<Credit compact>Some credit</Credit>)
    ).toMatchSnapshot()
})
