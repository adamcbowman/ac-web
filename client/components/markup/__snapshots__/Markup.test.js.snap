// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Br component 1`] = `
<div
  className="Main"
/>
`;

exports[`Br component 2`] = `
<div
  className="Ribbon"
/>
`;

exports[`Credit component 1`] = `
<span
  className="Credit"
  data-label="Credit"
>
  Some credit
</span>
`;

exports[`Credit component 2`] = `
<span
  className="Credit--Compact"
  data-label="Credit"
>
  Some credit
</span>
`;

exports[`Markup component 1`] = `
<div>
  Some content...
  <br />
</div>
`;
